includes/class-wc-cart.php

```diff
@@ -786,7 +786,7 @@ public function check_cart_item_stock() {
 			$held_stock     = wc_get_held_stock_quantity( $product, $current_session_order_id );
 			$required_stock = $product_qty_in_cart[ $product->get_stock_managed_by_id() ];
 
-			if ( $product->get_stock_quantity() < ( $held_stock + $required_stock ) ) {
+			if ( ! apply_filters( 'woocommerce_cart_has_enough_stock_for_product', $product->get_stock_quantity() >= ( $held_stock + $required_stock ), $product, $values ) ) {
```

Comment:
This sounds very confuse now, probably better invert the logic of this filter and also the name.

Reply:
Ok i will invert the logic. I was thinking for the filter name to be `woocommerce_cart_item_required_stock_is_not_enough`.